### DISCUSSION TITLE
Where are stored ssl key

### DISCUSSION THREAD

**Discussion by ghost** [NginxProxyManager/nginx-proxy-manager#2378] [Q&A]
Hello,
I love this product but i do have a question.

With the docker where are stored the ssl key ? (i ask because i do use NTS and Dns Over TLS/HTTPS and i would like to pass the position of the certificate to them (like that when NPM renew the key it's renewed for every services))

Thanks !

**Comment by KajKunstheim** [comment]
I would recommend using named volumes to store data for your docker container. Named volumes are stored in /var/lib/docker/volumes
Modify your docker-compose.yml to something like this:

`
version: "3"
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      # These ports are in format <host-port>:<container-port>
      - '80:80' # Public HTTP Port
      - '443:443' # Public HTTPS Port
      - '81:81' # Admin Web Port
      # Add any other Stream port you want to expose
      # - '21:21' # FTP
    environment:
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "npm"
      DB_MYSQL_NAME: "npm"
      # Uncomment this if IPv6 is not enabled on your host
      DISABLE_IPV6: 'true'
    volumes:
      - data:/data
      - letsencrypt:/etc/letsencrypt
    depends_on:
      - db

  db:
    image: 'jc21/mariadb-aria:latest'
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: 'npm'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'npm'
    volumes:
      - mysql:/var/lib/mysql
      
volumes:
  data:
  letsencrypt:
  mysql:
`

sorry the code-feature does not seem to work here. Indents were removed. Use this as reference: https://nginxproxymanager.com/setup/#using-mysql-mariadb-database

**Reply by ghost** [reply]
will try that and mark it as answer if it work thanks.
